python/lsst/ts/logging_and_reporting/web_app/services/scheduler_service.py

```diff
@@ -0,0 +1,768 @@
+        )
+        sphere_map.add_patches(
+            perimeter_df, patches_kwargs={"fill_color": None, "line_color": "black", "line_width": 1}
+        )
+
+    sphere_map.decorate()
+
+    sphere_map.add_marker(
+        ra=np.degrees(conditions.sun_ra),
+        decl=np.degrees(conditions.sun_dec),
+        name="Sun",
+        glyph_size=8,
+        circle_kwargs={"color": "yellow", "fill_alpha": 1},
+    )
+
+    sphere_map.add_marker(
+        ra=np.degrees(conditions.moon_ra),
+        decl=np.degrees(conditions.moon_dec),
+        name="Moon",
+        glyph_size=8,
+        circle_kwargs={"color": "orange", "fill_alpha": 0.8},
+    )
+
+    return plot
+
+
+def my_create_hpix_visit_map_grid(hpix_maps, visits, conditions, **kwargs):
+    """Create a grid of healpix maps with visits overplotted.
+
+    Notes
+    -----
+    Additional keyword args are passed to map_visits_over_hpix.
+
+    Parameters
+    ----------
+    map_hpix : `numpy.array`
+        An array of healpix values
+    visits : `pd.DataFrame`
+        The table of visits to plot, with columns matching the opsim database
+        definitions.
+    conditions : `rubin_scheduler.scheduler.features.Conditions`
+        An instance of a rubin_scheduler conditions object.
+
+    Returns
+    -------
+    plot : `bokeh.models.plots.Plot`
+        The plot with the map
+    """
+    # from schedview.plot.survey import map_visits_over_hpix
+    visit_map = {}
+    for band in hpix_maps:
+        visit_map[band] = my_map_visits_over_hpix(
+            visits.query(f"filter == '{band}'"), conditions, hpix_maps[band], **kwargs
+        )
+        visit_map[band].title = band
+
+    # Convert the dictionary of maps into a list of lists,
+    # corresponding to the rows of the grid.
+    num_cols = 3
+    map_lists = []
+    for band_idx, band in enumerate(hpix_maps):
+        if band_idx % num_cols == 0:
+            map_lists.append([visit_map[band]])
+        else:
+            map_lists[-1].append(visit_map[band])
+
+    map_grid = bokeh.layouts.gridplot(map_lists, sizing_mode="scale_both")
+
+    return map_grid
+
+
+def my_create_metric_visit_map_grid(
+    metric, metric_visits, visits, observatory, nside=32, use_matplotlib=False, **kwargs
+) -> UIElement | None:
+    """Create a grid of maps of metric values with visits overplotted.
+
+    Parameters
+    ----------
+    metric : `numpy.array`
+        An array of healpix values
+    metric_visits : `pd.DataFrame`
+        The visits to use to compute the metric
+    visits : `pd.DataFrame`
+        The table of visits to plot, with columns matching the opsim database
+        definitions.
+    observatory : `ModelObservatory`
+        The model observotary to use.
+    nside : `int`
+        The nside with which to compute the metric.
+    use_matplotlib: `bool`
+        Use matplotlib instead of bokeh? Defaults to False.
+
+    Returns
+    -------
+    plot : `bokeh.models.plots.Plot`
+        The plot with the map
```

Comment:
```suggestion
    Parameters
    ----------
    metric : `numpy.array`
        An array of healpix values.
    metric_visits : `pd.DataFrame`
        The visits to use to compute the metric.
    visits : `pd.DataFrame`
        The table of visits to plot, with columns matching the opsim database
        definitions.
    observatory : `ModelObservatory`
        The model observotary to use.
    nside : `int`
        The nside with which to compute the metric.
    use_matplotlib: `bool`
        Use matplotlib instead of bokeh? Defaults to False.

    Returns
    -------
    plot : `bokeh.models.plots.Plot` or `None`
        The plot with the map.
```